packages/tab/index.tsx

```diff
@@ -79,34 +114,41 @@ export default class Tab extends Component {
 
   get adapter() {
     return {
-      addClass: (className) => {
+      addClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.add(className);
         this.setState({classList});
       },
-      removeClass: (className) => {
+      removeClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.delete(className);
         this.setState({classList});
       },
-      hasClass: (className) => this.classes.split(' ').includes(className),
-      setAttr: (attr, value) => this.setState({[attr]: value}),
-      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
-      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
-      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
-      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      setAttr: (attr: keyof TabState, value?: string | boolean) => (
+        this.setState((prevState) => ({...prevState, [attr]: value}))
+      ),
+      getOffsetLeft: () =>
+        this.tabElement_.current && this.tabElement_.current.offsetLeft,
+      getOffsetWidth: () =>
+        this.tabElement_.current && this.tabElement_.current.offsetWidth,
+      getContentOffsetLeft: () =>
+        this.tabContentElement_.current &&
+        this.tabContentElement_.current.offsetLeft,
+      getContentOffsetWidth: () =>
+        this.tabContentElement_.current &&
+        this.tabContentElement_.current.offsetWidth,
       focus: () => this.tabElement_.current && this.tabElement_.current.focus(),
-      activateIndicator: (previousIndicatorClientRect) => this.setState({
-        activateIndicator: true,
-        previousIndicatorClientRect,
-      }),
+      activateIndicator: (previousIndicatorClientRect: ClientRect) =>
+        this.setState({
+          activateIndicator: true,
+          previousIndicatorClientRect,
+        }),
       deactivateIndicator: () => this.setState({activateIndicator: false}),
-      // computeIndicatorClientRect is redundant in mdc-tab and is going to be
-      // removed in another release
     };
   }
 
-  activate(computeIndicatorClientRect) {
+  activate(computeIndicatorClientRect?: ClientRect) {
```

Comment:
Why is this function a method while the others are instance fields?

Reply:
You mean why do we have activate on the class? I believe this is because tabbar needs this to be on the class instance. 

In tab bar's `componentDidMount`

```
if (this.tabList_[activeIndex]) {
      this.tabList_[activeIndex].activate({} /* previousIndicatorClientRect */);
}
```

This is similar to the vanilla version

Reply:
FYI: using lambdas instead of methods is against TypeScript Style:

> Classes should usually not contain properties initialized to arrow functions. Arrow function properties require the calling function to understand that the callee's this is already bound, which increases confusion about what this is, and call sites and references using such handlers look broken (i.e. require non-local knowledge to determine that they are correct). Code should always use arrow functions to call instance methods (const handler = (x) => this.listener(x);), and should not obtain or pass references to instance methods (const handler = this.listener; handler(x);).
> 
> Note: in some specific situations, e.g. when binding functions in a template, arrow functions as properties are useful and create much more readable code. Use judgement with this rule.

Reply:
Looks like you've got a lot of methods that are defined as lambdas and contain references to `this`.